Cleanup calls to dyn_cast

AndraBisca opened the pull request:
Cleanup unchecked calls to `llvm::dyn_cast` by replacing them with `llvm::cast`

abisca committed:
Change dyn_cast to cast where applicable: include/, lib/ for AIE and AIEX.

abisca committed:
Revert change on llvm::dyn_cast

abisca committed:
Format fixes

makslevental commented on include/aie/Dialect/AIE/IR/AIEOps.td:
```diff
@@ -1171,7 +1171,7 @@ def AIE_UseLockOp: AIE_Op<"use_lock", []> {
     }
     LockOp getLockOp() {
-      return llvm::cast<xilinx::AIE::LockOp>(getLock().getDefiningOp());
+      return llvm::dyn_cast<xilinx::AIE::LockOp>(getLock().getDefiningOp());
```
I think this one is backwards :)

AndraBisca replied:
I think I didn't mean to change that one actually. Good catch! :)

abisca committed:
Revert change

abisca committed:
Replaced dyn_cast in lib/Dialect/AIEVec

fifield committed:
Replace unchecked dyn_cast with cast in remaining files

Found and replaced 10 unchecked dyn_cast occurrences with cast:
- lib/Dialect/AIE/IR/AIEDialect.cpp (5 occurrences)
- lib/Dialect/AIE/Transforms/AIEObjectFifoRegisterProcess.cpp (1)
- lib/Targets/AIEFlowsToJSON.cpp (2)
- lib/Targets/AIETargetAirbin.cpp (3)
- lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp (1)

These were all cases where dyn_cast result was used without null checking,
meaning the code assumed the cast would always succeed.

fifield commented:
10/21/2025 - rebased and updated by copilot

fifield committed:
format
